enhancements/etcd/assisted-install.md

```diff
@@ -0,0 +1,221 @@
+efforts must be taken to minimize the risk of tolerating user defined unsafe
+cluster. For example to save on costs user chooses 2 master replicas.
+
+1.  Observe master replica count from install-config.
+1.  Observe install status and only tolerate less than 3 masters until install
+    complete. After install status is complete the operator will go into a
+degraded state until the cluster has 3 or more master nodes.
+1.  Observe infrastructure and only tolerate `BareMetal` type. Assumption being
+    that assisted installer will GA with support for only `BareMetal` (true?)
+
+### cluster-config-v1 configmap
+The expected master replica count provided by the `install-config` is a very
+important value for this design. This value will be used as the desired state
+for nodes during install. By using this value we can protect against allowing a cluster which
+expects 3 master nodes to be install complete if one of those nodes fails to
+pull ignition for example.
+
+cluster-config-v1 configmap with install-config embedded
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-config-v1
+  namespace: kube-system
+data:
+  install-config: |
+    apiVersion: v1
+    baseDomain: group-b.devcluster.openshift.com
+    controlPlane:
+      architecture: amd64
+      hyperthreading: Enabled
+      name: master
+      platform:
+        [..]
+      replicas: 3
+```
+
+Possible implementation: While immutable ConfigMaps are beta in 1.19 would it be
+possible to add the `immutable` field to `true` to this resource? `ImmutableEphemeralVolumes`
+feature gate is enabled by default in 1.19. This gives a little added protection to the
+validity of the data without additional complexity. If the feature is deprecated
+etcd team can own removal.
+
+### bootstrap configmap (no change)
+The `bootstrap` configmap in the `kube-system` namespace is where the value for
+`install status` is persisted. The design will observe this value as a
+condition. After the install status is complete the `cluster-etcd-operator` will no longer
+tolerate less than 3 master nodes and will go into a degraded state. Once the
+master count. If the configmap is missing or status is invalid the operator will
+go degraded. 
+
+bootstrap configmap
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  creationTimestamp: "2020-09-16T12:23:33Z"
+  name: bootstrap
+  namespace: kube-system
+data:
+  status: complete
+```
+
+### ConfigObservations
+Two news observations will be added to `cluster-etcd-operator` consumable by the
+controllers.
+
+```
+  spec:
+    logLevel: Normal
+    managementState: Managed
+    observedConfig:
+      install:
+        masterReplicas: 2
+        status: complete
+```
+
+#### ObserveInstallMasterReplicas
+This value will be read directly from the `cluster-config-v1` configmap.
+
+#### ObserveInstallStatus
+This value will be read directly from the `bootstrap` configmap.
+
+### Options 1: Extend checks in getEtcdEnvVars
```

Comment:
which one of the options is more aligned with current implementation of useUnsupportedUnsafeNonHANonProductionUnstableEtcd ? Since it's been working great so far